_rules/aria-required-accessible-name-gp8n89.md

```diff
@@ -0,0 +1,278 @@
+---
+id: gp8n89
+name: ARIA required accessible name
+rule_type: atomic
+description: |
+  This rule checks that WAI-ARIA accessible name is available when required.
+accessibility_requirements:
+  aria12:accessible_name_required:
+    title: ARIA 1.2, Accessible Name Required
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: satisfied
+  wcag20:1.3.1: # Info and Relationships (A)
+    secondary: This success criterion is **less strict** than this rule. Unlike WCAG, which might not mandate accessible names for certain elements like table elements, this rule requires them. Consequently, some of the failed examples satisfy this success criterion.
+  wcag20:2.4.4: # Link Purpose (In Context) (A)
+    secondary: This success criterion is **related** to this rule. This is because elements assigned the ARIA role of `link` necessitate an accessible name according to WCAG standards. Some of the examples that both passed and failed overlap with this success criterion.
+  wcag20:2.4.9: # Link Purpose (Link Only) (AAA)
+    secondary: This success criterion is **related** to this rule. This is because elements assigned the ARIA role of `link` necessitate an accessible name according to WCAG standards. Some of the examples that both passed and failed overlap with this success criterion.
+  wcag20:4.1.2: # Name, Role, Value (A)
+    secondary: This success criterion is **related** to this rule. This is because elements assigned an ARIA role corresponding to a user interface component necessitate an accessible name according to WCAG standards. Some of the examples that both passed and failed overlap with this success criterion.
+input_aspects:
+  - Accessibility Tree
+  - CSS styling
+  - DOM Tree
+  - ARIA
+acknowledgments:
+  authors:
+    - Giacomo Petri
+---
+
+## Applicability
+
+This rule applies to any [HTML or SVG element][] that is [included in the accessibility tree][] and has a [WAI-ARIA 1.2][wai-aria 1.2] [explicit semantic role][] that requires an [accessible name][] either from the [author](https://www.w3.org/TR/wai-aria-1.2/#namefromauthor) or the [content](https://www.w3.org/TR/wai-aria-1.2/#namefromcontent), unless the element [implicit semantic role][] matches its [explicit semantic role][].
```

Comment:
I'm not sure about the "no implicit" exception in that case.
It makes some sense for the `aria-*` attribute because native elements do some magic to auto set them up (e.g. `aria-checked` is set by `checked`, and `aria-expanded` can be some deep magic).
However in that case, there is no magic to set up the name, so a native `<img>` without a name is (also) an ARIA author eror and can be flagged by this rule.

Reply:
@Jym77, the rationale behind this is that certain roles necessitate an accessible name per ARIA requirements, even if it's not mandated by WCAG requirements. For instance, consider a search form: there's a search landmark wrapping a form (which needs an accessible name per ARIA specs), followed by a Search label, a search input with its associated label, and finally, a search submit button.

When using a screen reader, we'd hear "search" multiple times, which seems more like an error than an accessibility feature. This holds true for many roles. That's why I've specifically excluded implicit roles from the applicability.

It's worth noting that during today's ARIA meeting, @scottaohara brought up this topic, and it appears there's consensus on reviewing each role that requires an accessible name to determine whether it's always necessary or not (minutes available: https://github.com/w3c/aria/issues/2180).

I'd be more than willing to remove the exception with this change, but until then, I'm not comfortable doing so.

Reply:
I'm not entirely sure as to why this relates to implicit roles 🤔

Reply:
I'm not entirely sure as to why this relates to implicit roles 🤔 
If the search landmark wrapping a search field, or any of the other "bad" cases discussed there, is built with `<div role="…">` all over the place, we end up in the exact same situation, don't we?

In any case, it seems to be a case where something is currently an ARIA error without being a WCAG one (nor an accessibility issue). Which does require deciding between flagging ARIA error, or only flagging accessibility issues…

Even if ARIA group is working on it, I guess this won't happen before ARIA 1.3, so we should find a solution without it.

Whatever we decide, I think this is worth a Background note to explain why the rule is shaped that way, or the limits of its use.

Reply:
In my view, authors have the ability to utilize native semantic elements, such as a search element (`<search>`). If they do, even though it inherently carries the role of "search" as per ARIA, they may not be consciously utilizing ARIA. Conversely, if they consciously assign the role of search (`role="search"`), they are cognizant of ARIA requirements, which mandate the provision of an accessible name.

I'm not asserting that, in its current state, it does not constitute an ARIA violation (indeed it is), though it may not necessarily result in an accessibility issue. My concern lies in the potential confusion that may arise for those interpreting the rule if authors are required to adhere to this guideline for implicit roles as well.

However, again, I do acknowledge that, according to ARIA specs, it is indeed a requirement. Furthermore, once this concern is addressed by ARIA, there will no longer be a differentiation between implicit and explicit roles.

Therefore, I am inclined to remove the exception and update the background notes accordingly.

Reply:
Side note: there were already a background note about that :D

Reply:
@Jym77, I've made several updates:
- Expanded the applicability criteria to encompass any role, whether implicit or explicit.
- Adjusted the applicability to exclude conflicts with presentational roles. Previously, only explicit roles requiring acc name were included, so "none" and "presentation" were not part of the equation, excluding automatically presentational role conflicts. Now, considering the role of the element, also presentational roles causing conflicts apply, therefore I had to exclude them.
- Revised the background notes to address the discussed topic and emphasize that violating this ARIA rule does not necessarily breach WCAG standards.
- Enhanced the examples by incorporating implicit roles for both passed and failed examples, and removed implicit roles from inapplicable examples.
- Included a "non-failing" example into the inapplicable example of resolving presentational role conflicts, to clarify that this scenario is out of scope for this rule.